FamilySearchIosSampleApp/LoginVC.swift

```diff
@@ -0,0 +1,244 @@
+        let grant_type = "password";
+        
+        let params = "?username=" + username +
+            "&password=" + password +
+            "&grant_type=" + grant_type +
+            "&client_id=" + AppKeys.API_KEY;
+        
+        let urlAsString = tokenUrlAsString + params
+        
+        // create the post request
+        let request = NSMutableURLRequest(URL: NSURL(string: urlAsString)!)
+        request.HTTPMethod = "POST"
+        
+        let task = NSURLSession.sharedSession().dataTaskWithRequest(request){ data, response, error in
+            if (error != nil)
+            {
+                print("Error downloading token. Error: \(error)")
+                completionToken(responseToken: nil, errorToken: error)
+            }
+            else
+            {
+                do
+                {
+                    let jsonToken = try NSJSONSerialization.JSONObjectWithData(data!, options: .AllowFragments);
+                    if let token = jsonToken["access_token"] as? String
+                    {
+                        // parse the json to get the access_token, and save this token in NSUserDefaults
+                        let preferences = NSUserDefaults.standardUserDefaults()
+                        preferences.setValue(token, forKey: Utilities.KEY_ACCESS_TOKEN)
+                        preferences.synchronize()
+                    
+                        completionToken(responseToken: token, errorToken: nil)
+                    }
+                }
+                catch
+                {
+                    print("Error parsing token JSON. Error: \(error)");
+                }
+                
+            }
+        }
+        
+        task.resume()
+    }
+    
+    // get the user data
+    func getCurrentUserData(currentUserUrlString : String, accessToken : String, completionCurrentUser:(responseUser:User?, errorUser:NSError?) -> ())
+    {
+        let currentUserUrl = NSURL(string: currentUserUrlString);
+        
+        let configuration = NSURLSessionConfiguration.defaultSessionConfiguration();
+        let headers: [NSObject : AnyObject] = ["Accept":"application/json", "Authorization":"Bearer " + accessToken];
+        configuration.HTTPAdditionalHeaders = headers;
+        let session = NSURLSession(configuration: configuration)
+        
+        let currentUserTask = session.dataTaskWithURL(currentUserUrl!) { (data, currentUserResponse, errorUserData) in
+            // parse the currentUser data
+            do
+            {
+                let currentUserJson = try NSJSONSerialization.JSONObjectWithData(data!, options: .AllowFragments);
+                
+                if let usersJsonObject = currentUserJson["users"] as? [[String : AnyObject]]
+                {
+                    let user = User()
+                    let userJsonObject = usersJsonObject.first!
```

Comment:
Use a `guard` to extract the `first` object in the array rather than assume it exists.